### DISCUSSION TITLE
Add user expire date

### DISCUSSION THREAD

**Discussion by abrahamU** [open-webui/open-webui#7947] [Ideas]
Hi, nice work. I'm now using this application to run a local community in our Univercity, but I met a problem, since some people left without noticing us, I would like to have a expire data for user, so his account will be deactivated after this certain time. it's will be more easy for to manage my team.

thanks~

**Comment by max-bitons** [comment]
### +1 — Enterprise / on-prem appliance need

We operate a local Open WebUI deployment for an on-prem LLM appliance (multi-user, PostgreSQL, Redis). Account expiry is a recurring admin requirement:

- Contract / license end dates for external or temporary users
- Semester / project end for education and internal teams  
- Offboarding when people leave without notifying admins (same pain as the OP)

**Today’s workarounds are incomplete:**
- Manual role flip to `pending` does not scale
- Filter Functions can block chat after a custom date in `user.info`, but do **not** replace real deactivation or an Admin UI field
- SCIM/IdP expiry works only when SSO is in place; many air-gapped / LAN appliances stay on local accounts
- API keys have `expires_at`; user accounts do not

**Ask:** native per-user `expires_at` (or equivalent) with:
1. Admin Users UI: set / clear expiry
2. Automatic transition to `pending` (or inactive) when past expiry
3. Optional: store in DB + expose on admin/API for automation

Happy to help test a PR or clarify UX for appliance-style deployments. This would close a gap that currently forces every operator to invent a side process.
